dev/service-registry/service-locator/src/test/scala/com/lightbend/lagom/registry/impl/ServiceRegistryInteropSpec.scala

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (C) 2016-2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package com.lightbend.lagom.registry.impl
+
+import java.net.URI
+import java.util.{ Collections, Optional }
+
+import akka.actor.ActorSystem
+import akka.util.ByteString
+import com.lightbend.lagom.internal.javadsl.registry.{ RegisteredService => jRegisteredService, ServiceRegistryService => jServiceRegistryService }
+import com.lightbend.lagom.internal.scaladsl.registry.{ RegisteredService => sRegisteredService, ServiceRegistryService => sServiceRegistryService }
+import com.lightbend.lagom.javadsl.api.ServiceAcl
+import com.lightbend.lagom.javadsl.api.deser.{ MessageSerializer, StrictMessageSerializer }
+import com.lightbend.lagom.javadsl.api.transport.{ MessageProtocol, Method }
+import com.lightbend.lagom.javadsl.jackson.JacksonSerializerFactory
+import org.scalatest.concurrent.Futures
+import org.scalatest.{ FlatSpec, Matchers }
+import play.api.libs.json.{ Format, Json }
+
+class ServiceRegistryInteropSpec extends FlatSpec with Matchers with Futures {
+
+  val system = ActorSystem()
+  val jacksonSerializerFactory = new JacksonSerializerFactory(system)
+
+  behavior of "ServiceRegistry serializers"
+
+  it should "should interop betwen java and scala (RegisteredService)" in {
+    val msg = jRegisteredService.of("inventory", URI.create("https://localhost:123/asdf"), Optional.of("https"))
+    roundTrip(msg) should be(msg)
+  }
+
+  it should "should interop betwen java and scala when optional fields are empty (RegisteredService)" in {
+    val msg = jRegisteredService.of("inventory", URI.create("https://localhost:123/asdf"), Optional.empty[String])
+    roundTrip(msg) should be(msg)
+  }
+
+  it should "should interop betwen java and scala (ServiceRegistryService)" in {
+    val msg = jServiceRegistryService.of(URI.create("https://localhost:123/asdf"), Collections.singletonList(ServiceAcl.methodAndPath(Method.GET, "/items")))
+    roundTrip(msg) should be(msg)
+  }
+
+  it should "should interop betwen java and scala when optional fields are empty (ServiceRegistryService)" in {
+    val msg = jServiceRegistryService.of(URI.create("https://localhost:123/asdf"), Collections.emptyList[ServiceAcl])
+    roundTrip(msg) should be(msg)
+  }
```

Comment:
The four tests above exercise the following (with different payloads):

1. Given a java `J`
2. serialize to JSON
3. deserialize as a scala `S`
4. serialize back as JSON
5. deserialize as `J'`
6. assert `J' == J` 

I used that because at some point the payloads consumed/returned from the Service Registry javadsl were not compatible with the scaladsl client.